dhis-2/dhis-test-web-api/src/test/java/org/hisp/dhis/webapi/controller/DatastoreControllerTest.java

```diff
@@ -473,6 +477,78 @@ void testDeleteKeyJsonValue_ProtectedNamespaceWithSharing() {
     assertStatus(HttpStatus.OK, DELETE("/dataStore/pets/cat"));
   }
 
+  @Test
+  void testPutEntry_EntryDoesNotExistAndIsCreated() {
+    assertStatus(HttpStatus.CREATED, PUT("/dataStore/pets/emu", "{\"name\":\"harry\"}"));
+    JsonDatastoreValue emu = GET("/dataStore/pets/emu").content().as(JsonDatastoreValue.class);
+    assertEquals("harry", emu.getString("name").string());
+  }
+
+  @Test
+  void testPutEntry_EntryAlreadyExistsAndIsUpdated() {
+    assertStatus(HttpStatus.CREATED, PUT("/dataStore/pets/emu", "{\"name\":\"harry\"}"));
+    assertStatus(HttpStatus.OK, PUT("/dataStore/pets/emu", "{\"name\":\"barry\"}"));
```

Comment:
Nothing wrong with adding a bit of humor. Just that `harry` and `barry` is maybe not the best choice of values to show clearly what the difference is as they are very similar.

Reply:
😄 yep, i'll update it